### ISSUE TITLE
Arch-* targets not recognized by latest bash-completion

### ISSUE THREAD

**Issue by moneytoo** [SynoCommunity/spksrc#1078]
Hiting `<TAB>` after writing `make arch-` doesn't list any target when using latest `bash-completion` 2.1. For now it's fine in Debian stable as it uses 2.0 but 2.1 is already in testing and other distros already have it (Ubuntu for example).

It doesn't work because of [following rule](http://anonscm.debian.org/gitweb/?p=bash-completion/bash-completion.git;a=blob;f=completions/make;h=f8f71939b67640793142b7aac77846a3e88c6161;hb=afe39fd1e171d0ea3d4fb8ec0d8c8c14fa120ed8#l24) in bash-completion: 
`/^# Not a target/,/^$/        d;            # skip not target blocks` 

Running `make --print-data-base --dry-run` confirms the culprit:

```
...

# Not a target:
arch-88f6281:
#  Implicit rule search has not been done.
#  Modification time never checked.
#  File has not been updated.

...
```

Who's to blame? I miss any `make` know how so I don't know if the arch targets are defined correctly.
##

**Comment by astroganga** [comment]
I have the same problem on a virtual machine with debian 8 jessie. I need to continue use old vm with debian 7 to work around. Any news?

**Comment by Dr-Bean** [comment]
Kind of weird to not upgrade just because autocompletion isn't working for some make targets...Anyway, we can fix it with this: `.PHONY: $(addprefix arch-,$(AVAILABLE_ARCHS))`. We have a couple of other rules which could use similar changes.

**Comment by astroganga** [comment]
Actually I am using both vm. I have some uncommited code on debian 7 and I am preparing environment to transfer on debian 8. 
I tried your suggestion and it works. I think we need to add this too:
`.PHONY: $(addprefix publish-arch-,$(AVAILABLE_ARCHS))`

thanks

**Comment by Dr-Bean** [comment]
Problem is that it doesn't seem to actually work as intended, does it? It autocompletes, but it looks like the target is always considered up-to-date (...or something, haven't figured it out yet), because for subsequent commands, you end up with this:
```
/spksrc/spk/testpackage$ make arch-88f6281-5.2
make: Nothing to be done for 'arch-88f6281-5.2'.
```

Running `make arch-88f6281` works because that isn't considered part of the `.PHONY` target...but autocompletion probably won't ever work for that command.

**Comment by astroganga** [comment]
You're right. I was not aware of this serious side effect. 
Temporary I can comment the rule ` /^# Not a target/,/^$/ d; # skip not target blocks` but oviously it is not a solution.

**Comment by Dr-Bean** [comment]
See https://github.com/SynoCommunity/spksrc/pull/1919, which should solve the issue I believe. If you're thinking: "that looks like a mess"...Yes, it does ;)

**Comment by astroganga** [comment]
Hi,
excuse me for long wait, I had some personal problems.
I tried your fix #1919 and it works, but in subdirectories only. The root "spksrc" dir is still affected by the same problem. But it is a minor bug.
thanks

**Comment by Safihre** [comment]
Closing due to age, can be re-opened if someone wants to work on it again.
